integrations/mongodb_atlas/tests/test_document_store.py

```diff
@@ -192,7 +192,168 @@ def test_custom_embedding_field(self):
-                assert args[1]["$project"]["custom_vector"] == 1
 
         finally:
             database[collection_name].drop()
+
+    @pytest.mark.integration
+    def test_custom_content_field(self):
+        """Test that the custom content field is correctly used in the document store."""
+        # Create a document store with a custom content field
+        database_name = "haystack_integration_test"
+        collection_name = "test_custom_content_" + str(uuid4())
+
+        connection: MongoClient = MongoClient(
+            os.environ["MONGO_CONNECTION_STRING"], driver=DriverInfo(name="MongoDBAtlasHaystackIntegration")
+        )
+        database = connection[database_name]
+        if collection_name in database.list_collection_names():
+            database[collection_name].drop()
+        database.create_collection(collection_name)
+        database[collection_name].create_index("id", unique=True)
+
+        try:
+            custom_field_store = MongoDBAtlasDocumentStore(
+                database_name=database_name,
+                collection_name=collection_name,
+                vector_search_index="cosine_index",
+                full_text_search_index="full_text_index",
+                content_field="custom_text",
+            )
+
+            # Check that the content field is correctly set
+            assert custom_field_store.content_field == "custom_text"
+
+            # Write a document with standard content field
+            doc = Document(content="test content")
+            custom_field_store.write_documents([doc])
+
+            # Verify it's stored with the custom field name in MongoDB
+            database_doc = database[collection_name].find_one({"id": doc.id})
+            assert "custom_text" in database_doc
+            assert database_doc["custom_text"] == "test content"
+            assert "content" not in database_doc
+
+            # Retrieve the document and verify it has the standard content field
+            retrieved_docs = custom_field_store.filter_documents()
+            assert len(retrieved_docs) == 1
+            assert retrieved_docs[0].content == "test content"
+
+            # This is a mock test for text search
+            with patch.object(custom_field_store, "_collection") as mock_collection:
+                # Setup the mock
+                mock_collection.aggregate.return_value = []
+
+                # Execute the method
+                custom_field_store._fulltext_retrieval(query="test query")
+
+                # Verify that the text search is using the standard content path
+                # Note: The content path in _fulltext_retrieval is hardcoded to "content"
+                args = mock_collection.aggregate.call_args[0][0]
+                assert args[0]["$search"]["compound"]["must"][0]["text"]["path"] == "content"
+
+        finally:
+            database[collection_name].drop()
+
+    @pytest.mark.integration
+    def test_document_conversion_methods(self):
```

Comment:
I would transform this integration test into a unit test

Reply:
fair fair